src-graphql/client.js

```diff
@@ -24,11 +24,11 @@ function fetchAllProductResources(product, client) {
   const images = product.images;
   const variants = product.variants;
 
-  if (images && images[images.length - 1].hasNextPage.valueOf()) {
+  if (images && images.length && images[images.length - 1].hasNextPage.valueOf()) {
```

Comment:
Scalars aren't being unwrapped properly by `Boolean` for some reason. I think @minasmart was looking into this.